File: typescript-service/functions/create.ts
import { faas, documents } from "@nitric/sdk";
import { Example } from "../common";
import { uuid } from "uuidv4";

// Start your function here
faas.start(async (request: faas.NitricTrigger<Example>): Promise<string> => {
  const example = request.dataAsObject();

  // generate a new uuid
  const id = uuid();

  // Create a new example document
  await documents().collection("example").doc(id).set(example);

  return `Created example with ID: ${id}`;
});

File: python-stack/functions/create.py
from nitric import faas
from nitric.api import Documents
import json

from common.example import generate_id


async def handler(ctx: faas.HttpContext) -> faas.HttpContext:
    example = json.loads(ctx.req.body)
    doc_id = generate_id()
    await Documents().collection("examples").doc(doc_id).set(example)

    ctx.res.body = f'Created example with ID: {doc_id}'
    return ctx


if __name__ == "__main__":
    faas.http(handler).start()
